### PR TITLE
SLS-507 Change Python tests to use PALI.

### PR DESCRIPTION
**Pull request by ddanderson**
- added facility for setting individual extension args easily.
- Fixed bug in PALM that resulted in missing the main page.

### CODE REVIEW

File: test/suite/test_oligarch04.py [hunk 1]
```diff
@@ -70,7 +61,7 @@ def test_oligarch04(self):
             cursor["Hi " + str(i)] = "There"
             cursor["OK " + str(i)] = "Go"
             if i % 10000 == 0:
-                time.sleep(1)
+                time.sleep(5)
```
Comment (pmacko86) [inline test/suite/test_oligarch04.py:64]: Why is this necessary?

**Review by pmacko86 — APPROVED** [review]
LGTM, with a question.

File: test/suite/test_oligarch04.py (hunk 1 above)
Comment (ddanderson) [reply test/suite/test_oligarch04.py:64]: That's a very good question - I do know that if the delay is not there, the count at the end is incorrect (I guess some items aren't available).  hmmm - these delays have been in the tests for a while.  I'll ping our slack channel.

Comment (pmacko86) [reply test/suite/test_oligarch04.py:64]: Can you please create a ticket that describes this failure?
